ra/ra.go

```diff
@@ -714,18 +755,122 @@ func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, names []str
 	return nil
 }
 
+// FinalizeOrder accepts a request to finalize an order object and, if possible,
+// issues a certificate to satisfy the order. If an order does not have valid,
+// unexpired authorizations for all of its associated names an error is
+// returned. Similarly we vet that all of the names in the order are acceptable
+// based on current policy and return an error if the order can't be fulfilled.
+func (ra *RegistrationAuthorityImpl) FinalizeOrder(ctx context.Context, req *rapb.FinalizeOrderRequest) error {
```

Comment:
Note to self: I forgot to check that the order isn't expired here. Will address shortly with a fix & a test.

Reply:
I ended up doing this in the WFE.